elobot.py

```diff
@@ -11,195 +11,225 @@
+        """Send a heartbeat if necessary"""
         now = int(time.time())
         if now > self.last_ping + 3:
             self.slack_client.server.ping()
             self.last_ping = now
 
     def talk(self, message):
-        #print(message)
-        self.slack_client.api_call('chat.postMessage', channel=self.channel, text=message, username=self.config['bot_name'])
+        """Send a message to the Slack channel"""
+        self.slack_client.api_call('chat.postMessage', channel=self.channel_id, text=message, username=self.name)
 
-    def run(self):
-        print('running!')
-        #self.talk(self.config['bot_name'] + ' online!')
+    def talk_to(self, user_id, message):
+        message = message[0].lower() + message[1:]
+        self.talk(f'<@{user_id}>, {message}')
 
+    def run(self):
         while True:
-            if self.slack_client.server.connected:
-                messages = self.slack_client.rtm_read()
-                for message in messages:
-                    if message.get('type', False) == 'message' and message.get('channel', False) == self.channel and message.get('text', False):
-                        print('processing message in my channel:\n{}'.format(message))
-                        if WINNER_REGEX.match(message['text']):
-                            self.winner(message)
-                        elif CONFIRM_REGEX.match(message['text']):
-                            self.confirm(message['user'], message['text'])
-                        elif CONFIRM_ALL_REGEX.match(message['text']):
-                            self.confirm_all(message)
-                        elif DELETE_REGEX.match(message['text']):
-                            self.delete(message['user'], message['text'])
-                        elif LEADERBOARD_REGEX.match(message['text']):
-                            self.print_leaderboard()
-                        elif UNCONFIRMED_REGEX.match(message['text']):
-                            self.print_unconfirmed()
-                self.heartbeat()
-                time.sleep(0.1)
-            else:
-                #Attempt to reconnect if failed to read from websocket
-                self.connect()
-                continue
-
-    def winner(self, message):
-        # 0: space, 1: winning verb, 2: loser_id, 3: first score, 4: second score
-        # then 0 or more of...
-        # 5: 2nd game hyphenated score, 6: 2nd game first score, 7: 2nd game second score
-        msg = message['text']
-        values = re.split(WINNER_REGEX, msg)
-        if not values or len(values) < 5:
-            return
-
-        loser_id = values[2]
-
-        # csv game list starts after the end of the slack username
-        games_csv = msg[(msg.index('>') + 1):]
-        games = games_csv.replace(' ', '').split(',')
-
-        for game in games:
-            scores = game.split('-')
-            if len(scores) != 2:
-                continue
+            time.sleep(0.1)
+            self.heartbeat()
```

Comment:
why `heartbeat` before `ensure_connected`...?

Reply:
Shouldn't matter either way, should it?